sandbox/apparmor/apparmor_test.go

```diff
@@ -314,110 +312,131 @@ func (s *apparmorSuite) TestProbeAppArmorKernelFeatures(c *C) {
 	}
 }
 
-func (s *apparmorSuite) TestProbeAppArmorParserFeatures(c *C) {
-	var features = []string{"unsafe", "include-if-exists", "qipcrtr-socket", "mqueue", "cap-bpf", "cap-audit-read", "xdp", "userns", "unconfined", "prompt"}
-	// test all combinations of features
-	for i := 0; i < int(math.Pow(2, float64(len(features)))); i++ {
-		expFeatures := []string{}
-		d := c.MkDir()
-		contents := ""
-		var expectedCalls [][]string
-		for j, f := range features {
-			code := 0
-			if i&(1<<j) == 0 {
-				expFeatures = append([]string{f}, expFeatures...)
-			} else {
-				code = 1
-			}
-			expectedCalls = append(expectedCalls, []string{"apparmor_parser", "--preprocess"})
-			contents += fmt.Sprintf("%d ", code)
+func probeOneParserFeature(t *testing.T, known *[]string, parserPath, featureName, profileText string) {
+	t.Helper()
+
+	const script = `#!/bin/sh
+set -e
+test "$(cat | tr -d '\n')" = '%s'
+`
+	if err := os.WriteFile(parserPath, []byte(fmt.Sprintf(script, profileText)), 0o755); err != nil {
+		t.Fatal(err)
+	}
+
+	cmd, _, _ := apparmor.AppArmorParser()
+	if cmd.Path != parserPath {
+		t.Fatal("Unexpectedly using apparmor parser from", cmd.Path)
+	}
+
+	features, err := apparmor.ProbeParserFeatures()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(features) != 1 || features[0] != featureName {
+		t.Error("Unexpected features", features, "for profile", profileText)
+	}
+
+	*known = append(*known, featureName)
+}
+
+func TestProbeParserFeatures(t *testing.T) {
```

Comment:
hm ok in general, but the package is already using `check`. This will be inconsistent now if one wants to run for instance single test through `-check.f` or `-run`. I'm not opposed to using plain testing.T but perhaps let's leave it as it is, and discuss the option of using plain testing.T in new packages we introduce?

Reply:
I started implementing this with check and ended up with a separate suite, lack of defer/cleanup across functions or lots of repetition in test code to avoid splitting.

I can rewrite this with check but ... maybe this is fine? I really don't think we should stay married to check in cases where doing so is not very difficult.